tests/Task.test.ts

```diff
@@ -1290,7 +1291,7 @@ describe('check removal of the global filter', () => {
         ({ globalFilter, markdownTask, expectedDescription }) => {
             // Arrange
             if (globalFilter != '') {
-                updateSettings({ globalFilter: globalFilter });
+                updateSettings({ globalFilter: new GlobalFilter(globalFilter) });
```

Comment:
As an example of what an abstraction might look like - this could become something like this - if we wanted the global filter to be static.

```ts
updateSettings({ globalFilter: new GlobalFilter(globalFilter) });
```

Would instead become the following...

```ts
GlobalFilter.setFilter(globalFilter);
```

Internally, `GlobalFilter.setFilter()` would do the `updateSettings()` call - but that would be invisible to users.

And there could also be other methods line:

```ts
GlobalFilter.removeFilter();
GlobalFilter.matches(line); // etc
```

---

The important thing is that the code that uses `GlobalFilter` knows nothing about where the filter value comes from. That is all "encapsulated" - that is, totally hidden away.

This means that the function `getGlobalFilter()` would not needed at all...

I hope that explains the idea a bit better...

Reply:
yup, I thought about making it static but wasn't sure that it was the path you preferred as well if it was ok to "hide" the calls of `getSettings()` & `updateSettings()` in this class. But we are aligned now. Thanks for your comments!

Reply:
Yeah, it's all about 'syntactic sugar' - hiding away the details.

Anticipating what might happen, suppose you find two different bits of code that do the 'search a task line for the global filter' differently, like maybe one area does a 'simple substring search', and another area does a 'do a regex search', there would need to be two different clearly named methods on GlobalFilter that indicate how they differ...

So that we can be sure that the behaviour is unchanged. But at least the different approaches would all be present in the one location.

Oooohhh - I believe that the regex approach is in the auto-suggest code... FYI, there is an active PR on that right now: #1773 - no idea if what you are doing would create any conflicts. If so, you might wish to restart the GlobalFilter once the above PR is merged....

Reply:
Using `static` in this case is great indeed: all the ugly code like `const { globalFilter } = getSettings();` is gone now.

This actually made me think that a good idea would be actually reworking the `Settings.ts`: there should be a `Settings` class and the current interface shall be something like `ISettings`. This way the access to all the settings would be much prettier not to mention the C-styled `resetSettings()` would be a much more readable `Settings.reset()`.

As for the other PR, I have no problem merging my code afterwards.

Reply:
Thank you. 

So let’s discuss this next time we chat. The use of statics may be a step on the way to something better later on, not the end point.